api/src/test/java/com/datastrato/gravitino/rel/TestPartitions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.rel.expressions.literals.Literal;
+import com.datastrato.gravitino.rel.expressions.literals.Literals;
+import com.datastrato.gravitino.rel.partitions.Partition;
+import com.datastrato.gravitino.rel.partitions.Partitions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import java.time.LocalDate;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TestPartitions {
+
+  @Test
+  public void testPartitions() {
+    Partition partition =
+        Partitions.range("p0", Literals.NULL, Literals.integerLiteral(6), Maps.newHashMap());
+    Assertions.assertEquals("p0", partition.name());
+    Assertions.assertEquals(Maps.newHashMap(), partition.properties());
+    Assertions.assertEquals(Literals.NULL, ((Partitions.RangePartition) partition).upper());
+    Assertions.assertEquals(
+        Literals.integerLiteral(6), ((Partitions.RangePartition) partition).lower());
+
```

Comment:
I think it would be better if you provided a complete example of creating a partitioned table.

Reply:
This PR focuses on the definition of partitions and partition management, while examples of creating partitioned tables will be provided in another PR.